test/framework/cluster.go

```diff
@@ -230,6 +231,16 @@ func PersistentCluster() ClusterE2ETestOpt {
 	}
 }
 
+// WithSkipRegenerateClusterConfig skips generating the cluster config if it finds
+// a cluster config already in the ClusterConfigLocation. Prevents overriding an
+// existing updated cluster config with the default one generated by the CLI
+// `generate clusterconfig` command
+func WithSkipRegenerateClusterConfig() ClusterE2ETestOpt {
+	return func(e *ClusterE2ETest) {
+		e.SkipRegenerateClusterConfig = true
```

Comment:
We can have this as a follow up since it's a small thing, but should we just make this the default behavior and avoid having to pass a new option?

Reply:
Wait I see we already had this as default. So now we are changing the default to be the opposite, it will overwrite by default?

Reply:
The reason I had to do this is, for tests calling the `WithClusterConfig` method, a cluster config gets generated by that method with the proper fillers and options. Then when we go to the simple flow method, it again calls on the GenerateClusterConfig method that directly runs the CLI command without the fillers and overrides the previously created correct cluster config.

In case of the Ubuntu 2204 tests, this was causing the correct Ubuntu-based cluster config to be replaced by a `bottlerocket` os family in the machine configs since that's the default generated by the CLI command that runs as part of the simple flow method.

Reply:
Removed this new option and instead created separate functions for the new tests that don't generate the cluster config